common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -551,44 +551,27 @@ private boolean runShutdownHooks() {
             return terminationFuture();
```

Comment:
(for the line above): minor optimization ... `isShuttingDown()` is called above which will use the state, and you are also getting the state below in a loop. you could get the state outside the loop, use it above, and then refresh it at the end of the loop.

```java
int oldState = STATE_UPDATER.get(this);
if (isShuttingDown(state)) {
  return terminationFuture();
}

for(;;) {
//try atomic set
  oldState = STATE_UPDATER.get(this);
}

@Override
public boolean isShuttingDown() {
  return isShuttingDown(STATE_UPDATER.get(this));
}

private static boolean isShuttingDown(int state) {
  return state >= ST_SHUTDOWN;
}
```